### PR TITLE
add main card component and drag-and-drop for tags

### PR DESCRIPTION
**Pull request by NadiaKishchuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/features/AdminPage/MainBlock/DragableTags/ClickableTagItem.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useState } from 'react';
+import './DragableTags.styles.scss';
```
Comment (NadiaKishchuk) [inline src/features/AdminPage/MainBlock/DragableTags/ClickableTagItem.tsx:2]: this file  is temporary, I will use this code than in correct place in addnewstreetcode page

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    </Input.Group>
+                )
+                    : (
+                        <Form.Item label="Назва події">
+                            <Input />
+                        </Form.Item>
+                    )}
+
+                <DragableTags setTags={setSelectedTags} tags={selectedTags} />
+                <Select style={{ width: '100%' }} mode="tags" onSelect={onSelectTag} onDeselect={onDeselectTag}>
```
Comment (Katerix) [inline src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:64]: are inline styles necessary here?

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        setSelectedTags(selectedTags.filter((t) => t.title !== value));
+    };
+    return (
+        <Form layout="vertical" className="mainblock-add-form">
+            <>
+                <Form.Item label="Номер стріткоду" style={{ fontSize: '20px' }} className="mainblock-add-form-item">
+                    <InputNumber min={0} />
+                </Form.Item>
+                {streetcodeType === 'people' ? (
+                    <Input.Group compact style={{ fontSize: '20px' }}>
```
Comment (Katerix) [inline src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:45]: and here?

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+        setSelectedTags([...selectedTags, { ...selected, visible: true }]);
+    };
+    const onDeselectTag = (value:string) => {
+        setSelectedTags(selectedTags.filter((t) => t.title !== value));
+    };
+    return (
+        <Form layout="vertical" className="mainblock-add-form">
+            <>
+                <Form.Item label="Номер стріткоду" style={{ fontSize: '20px' }} className="mainblock-add-form-item">
```
Comment (Katerix) [inline src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:41]: same

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx (hunk 3 above)
Comment (NadiaKishchuk) [reply src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:45]: no, I forgot to remove this, thanks)

File: src/features/AdminPage/MainBlock/DragableTags/ClickableTagItem.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { Button } from 'antd';
+
+import TagItem from '@/app/common/components/Tag/TagItem.component';
+import { TagVisible } from '@/models/additional-content/tag.model';
+
+const ClickableTagItem: React.FC<{ tag: TagVisible }> = ({ tag }) => {
+    const [curTag, setTag] = useState<TagVisible>(tag);
+    const onClick = (e) => {
+        console.log(curTag);
```
Comment (Tysyatsky) [inline src/features/AdminPage/MainBlock/DragableTags/ClickableTagItem.tsx:11]: is console.log necessary?

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+import DragableTags from './DragableTags/DragableTags.component';
+
+const MainBlockAdmin: React.FC<{ streetcodeType: 'people' | 'event' }> = ({ streetcodeType }) => {
+    const [tags, setTags] = useState< Tag[]>([
+        { id: 1, title: 'first' },
+        { id: 2, title: 'second' },
+        { id: 3, title: 'third' },
+    ]);
+
```
Comment (Tysyatsky) [inline src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:19]: is it tags for testing?

File: src/features/AdminPage/MainBlock/DragableTags/DragableTags.component.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                                {(provided: DraggableProvided, snapshot: DraggableStateSnapshot) => (
+                                    <div
+                                        className="draggable-tag"
+                                        ref={provided.innerRef}
+                                        {...provided.draggableProps}
+                                        {...provided.dragHandleProps}
+                                        style={getItemStyle(snapshot.isDragging, provided.draggableProps.style)}
+                                    >
+
+                                        <ClickableTagItem
```
Comment (Tysyatsky) [inline src/features/AdminPage/MainBlock/DragableTags/DragableTags.component.tsx:62]: are these spaces necessary?

File: src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx (hunk 6 above)
Comment (NadiaKishchuk) [reply src/features/AdminPage/MainBlock/MainBlockAdmin.component.tsx:19]: yes, later I will get it from db

**Pull request merged**